app/providers/atmosphere/optimization_strategy/default.rb

```diff
@@ -90,8 +90,17 @@ class VmtAndFlavorAndTenant
         include Atmosphere::Utils
 
         def initialize(tmpls, appliance, options={})
-          @tmpls = tmpls
-          @appliance = appliance
+          @tmpls = VirtualMachineTemplate.where(id: tmpls.map(&:id))
+            .includes(tenants: {virtual_machine_flavors: [:os_families, {flavor_os_families: :os_family}]})
```

Comment:
Line is too long. [107/80]<br>Place the . on the previous line, together with the method call receiver.<br>Align the operands of an expression in an assignment spanning multiple lines.<br>Space inside { missing.<br>Space inside } missing.